include/deal.II/matrix_free/matrix_free.h

```diff
@@ -628,19 +628,6 @@ class MatrixFree : public Subscriptor
          const std::vector<QuadratureType> &                    quad,
          const AdditionalData &additional_data = AdditionalData());
 
-  /**
-   * Initializes the data structures. Same as above, but  using hp::DoFHandlers.
-   *
-   * @deprecated Use the overload taking a DoFHandler object instead.
-   */
-  template <typename QuadratureType, typename number2>
-  DEAL_II_DEPRECATED void
-  reinit(const Mapping<dim> &                                   mapping,
-         const std::vector<const hp::DoFHandler<dim> *> &       dof_handler_hp,
-         const std::vector<const AffineConstraints<number2> *> &constraint,
-         const std::vector<QuadratureType> &                    quad,
-         const AdditionalData &additional_data = AdditionalData());
```

Comment:
We cannot do this. I only have deprecated these after the last release.

Reply:
Correct.

Reply:
Yes I know, but otherwise the CI complains about using the deprecated `hp::DoFHandler` :(

I thought that technically, this wouldn't make a difference. As the base class of `hp::DoFHandler` is actually the basic `DoFHandler`. So when using a `hp::DoFHandler`, we would now automatically invoke the `DoFHandler` equivalent. Code like this compiles: https://godbolt.org/z/do6znG

Reply:
The problem is the case `std::vector<B *> bar{&b, &b};`.

Reply:
I would really love to see 
```
namespace hp
{
  template<int dim, int spacedim>
  DEAL_II_DEPRECATED using DoFHandler = dealii::DoFHandler<dim, spacedim>;
}
```
After all the changes this might already work?

Reply:
I see the problem now. I'll try to find a workaround to use the `reinit` function without triggering the deprecation warning.

With your suggestion for the alias I get
```
error: conflicting declaration of template ‘template<int dim, int spacedim> using DoFHandler = dealii::DoFHandler<dim, spacedim>’
```
Even in the hp namespace. Maybe we should leave it as is.

Reply:
I mean as soon as hp::DoFHandler is deprecated all functions using it are implicitly deprecated. So it would probably be fine to remove the `[[deprecated]]` attribute from all problematic functions if that helps.

Reply:
CI is particularly complaining about using deprecated arguments in these function, not about the deprecation of the functions themselves.

Affected examples are as far as I can tell: 37, 48, 50, 59, 67

Testers complain with errors like:
```
error: 'DoFHandler<2, 2>' is deprecated
```
Using `gcc 8.1` without the `-Werror` option:
```
warning: ‘template<int dim, int spacedim> class dealii::hp::DoFHandler’ is deprecated
```